data-prepper-plugins/http-source/src/main/java/org/opensearch/dataprepper/plugins/source/loghttp/HTTPSourceConfig.java

```diff
@@ -134,6 +135,10 @@ public int getRequestTimeoutInMillis() {
         return requestTimeoutInMillis;
     }
 
+    public int getBufferTimeoutInMillis() {
+        return (int)(BUFFER_TIMEOUT_FRACTION * requestTimeoutInMillis);
```

Comment:
With this change, we might want to make a different minimum for the `request_timeout` value. This value will return `0` if the `requestTimeoutInMillis` is 1, right?

Reply:
Yes, it will be 0. Is the concern that a value of `0` may disable the buffer timeout?

Reply:
Yes, but maybe that is ok?

Reply:
It might be OK for our current buffer implementation but this may change. So I updated the minimum validation for `request_timeout` to 2 to avoid any confusion.